server.py

```diff
@@ -9,12 +9,11 @@
 app = Bottle()
 
 questions = {}
-solved_questions = defaultdict(set)
-submissions = defaultdict(list)  # dictionary for storing the usernames
+submission_record = defaultdict(list)  # dictionary for storing the usernames
 question_dir = 'files/questions'
 
 Question = namedtuple('Question', 'output statement')
-Submission = namedtuple('Submission', 'question time result output')
+Submission = namedtuple('Submission', 'question time output result')
```

Comment:
might be easier to say `is_correct` instead of `result` no? then whenever it's used in conditions it can be used as `if attempt.is_correct` instead of `if attempt.result` which tells me nothing about what is the "meaning" of the thing being checked.